### PR TITLE
ID22968:Get list of available sentinel 1 and 2 images

### PR DESCRIPTION
**Pull request by sviridovartem**

### CODE REVIEW

**Comment by sviridovartem** [comment]
`sentinelhub_images_creds.json`
should be like this
`{
  "client_id": "d4b27d63-67d7-4371-b3d2-a6de50c32fb1",
  "client_secret": "md&1tRdVv6+s>F&6x9tn0u4Es*/J]mp+tan0~c85"
}`
these creds I created for testing, where can I get corporate creds, or just create a new one for the company?

File: webapplication/aoi/models.py [hunk 1]
```diff
@@ -20,6 +25,8 @@ class AoI(models.Model):
     polygon = models.PolygonField(spatial_index=True, verbose_name='Polygon')
     createdat = models.DateTimeField(default=timezone.now)
     type = models.IntegerField(choices=AREA_TYPE_CHOICES, default=USER_DEFINED_TYPE)
+    available_dates = models.JSONField(blank=True, null=True, verbose_name='Available dates')
```
Comment (adolgarev) [inline webapplication/aoi/models.py:28]: it is better to make clear available dates of what

File: webapplication/aoi/models.py [hunk 2]
```diff
@@ -54,6 +136,14 @@ class Component(models.Model):
...
         (DATE_DAY_TYPE, "DAY")
     )
+    SENTINEL_IMAGE_NONE_TYPE = 1
+    SENTINEL_IMAGE_SENTINEL1_TYPE = 2
+    SENTINEL_IMAGE_SENTINEL2_TYPE = 3
+    SENTINEL_IMAGE_CHOICES = (
+        (SENTINEL_IMAGE_NONE_TYPE, "None"),
+        (SENTINEL_IMAGE_SENTINEL1_TYPE, "Sentinel-1"),
+        (SENTINEL_IMAGE_SENTINEL2_TYPE, "Sentinel-2"),
+    )
```
Comment (adolgarev) [inline webapplication/aoi/models.py:146]: Can there be both? Some other type? Do all models so far belong to only one of these types?

File: webapplication/aoi/views.py [hunk 3]
```diff
@@ -40,7 +40,9 @@ def get_queryset(self):
         return queryset.filter(user=self.request.user)
         
     def create(self, request, *args, **kwargs):
-        serializer = self.get_serializer(data=request.data)
+        request_data = request.data.copy()
+        request_data["available_dates"] = AoI.get_available_image_dates(request_data['polygon'], short_period=True)
```
Comment (adolgarev) [inline webapplication/aoi/views.py:44]: This should be done in background

**Review by adolgarev — COMMENTED** [review]
One cannot afford running get_available_image_dates in a view. Thus we should

1. Store in the AOI model the last time dates were updated, by default it is NULL.
2. Run script in foreground which every 10 seconds or so gets from db all AOI with dates either NULL or past 1 week. The same way webapplication/aoi/management/commands/notebook_executor.py is done.
3. On FE after AOI is created re-fetch it every 5 seconds until dates are set. Also if dates are not set yet during report creation warn user about this.

Also move th logic to the command itself, lets not put everything to the model

File: webapplication/aoi/models.py (hunk 2 above)
Comment (sviridovartem) [reply webapplication/aoi/models.py:146]: Michael Yushchuk
могут быть 2
могут быть другие в теории
но пока не предвидится

File: webapplication/aoi/models.py (hunk 1 above)
Comment (sviridovartem) [reply webapplication/aoi/models.py]: updated

File: webapplication/aoi/models.py (hunk 2 above)
Comment (sviridovartem) [reply webapplication/aoi/models.py]: added Sentinel-1 & Sentinel-2

File: webapplication/aoi/views.py (hunk 3 above)
Comment (sviridovartem) [reply webapplication/aoi/views.py]: moved in crontab task

File: webapplication/aoi/models.py [hunk 4]
```diff
@@ -69,6 +72,7 @@ class Component(models.Model):
     success = models.BooleanField(default=False, verbose_name='Validation succeeded')
     additional_parameter = models.CharField(max_length=50, null=True, blank=True, verbose_name='Additional parameter')
     run_on_gpu = models.BooleanField(default=True, verbose_name='Whether GPU is needed for a component to run')
+    sentinel_image_type = models.IntegerField(choices=settings.SENTINEL_IMAGE_CHOICES, default=settings.SENTINEL_IMAGE_NONE_TYPE)
```
Comment (adolgarev) [inline webapplication/aoi/models.py:75]: I believe we should rely on required creds. E.g., if sentinel1_aws_creds_required then Sentinel-1 is required. We have an issue that remote executor doesn't require credentials. I'd better set required creds for remote executor even if they are not set.

File: webapplication/aoi/models.py [hunk 5]
```diff
@@ -69,6 +72,7 @@ class Component(models.Model):
     success = models.BooleanField(default=False, verbose_name='Validation succeeded')
     additional_parameter = models.CharField(max_length=50, null=True, blank=True, verbose_name='Additional parameter')
     run_on_gpu = models.BooleanField(default=True, verbose_name='Whether GPU is needed for a component to run')
+    sentinel_image_type = models.IntegerField(default=settings.SENTINEL_IMAGE_NONE_TYPE)
```
Comment (adolgarev) [inline webapplication/aoi/models.py:75]: You should compute and return a set of flags what kind of imagery is required in a view. No need to store this in db.

File: webapplication/sip/settings.py [hunk 6]
```diff
@@ -258,3 +262,8 @@
 
 DEFAULT_SYSTEM_NOTIFICATION_EMAIL = ""
 DEFAULT_TRANSACTION_ERROR = "Something went wrong, please contact us"
+
+SENTINEL_IMAGE_NONE_TYPE = 1
+SENTINEL_IMAGE_SENTINEL1_TYPE = 2
+SENTINEL_IMAGE_SENTINEL2_TYPE = 3
+SENTINEL_IMAGE_SENTINEL1_2_TYPE = 4
```
Comment (adolgarev) [inline webapplication/sip/settings.py:269]: The view should calculate and return a set of flags. E.g. ComponentSerializer should return something like "sentinels": ["Sentinel1", "Sentinel2"].

Comment (sviridovartem) [reply webapplication/sip/settings.py]: updated

**Pull request merged**